### DISCUSSION TITLE
Exposing Types

### DISCUSSION THREAD

**Discussion by rjerue** [vercel/next.js#13163] [Help]
Hello,

There's a lot of types that Next does export pretty clearly. It looks like the pattern for this is typically an index file in the root that does an `export * from 'dist/something/index'`. Not all the files, particularly those under `dist/next/server` are exposed. If I want to get `Server` then I have to do `import Server from 'next/dist/next-server/server/next-server';` Is that how most folks are doing it?

You can also leverage ReturnType, but it'd be nice if there was a consistent way of importing types.

**Comment by timneutkens** [comment]
Why do you need the server type? It's not a public api and can change at any time.

**Reply by rjerue** [reply]
I'm doing something like this...

```ts
// foo.ts
import next from 'next';

const nextServer = next(...)

export const nextServer;

// bar.ts
function prepareNextServer(nextServer: TypeGoHere){
    // Do something with nextServer
    // Particularly usage of nextServer.prepare, nextServer.getRequestHandler, and nextServer.render
}
```

Each of the above functions are [used in the docs](https://nextjs.org/docs/advanced-features/custom-server). Totally understandable that everything in the server types is not public, but is there a type that represents the slice of things that are?
